app/controllers/Checkout.scala

```diff
@@ -57,7 +59,16 @@ object Checkout extends Controller with LazyLogging {
 
     touchpointBackendResolution.backend.checkoutService.processSubscription(formData, idUserOpt).map { case CheckoutResult(_, userIdData, subscription) =>
       val passwordForm = userIdData.toGuestAccountForm
-      Ok(view.thankyou(subscription.name, formData.personalData, passwordForm, touchpointBackendResolution))
+
+      val ratePlan = touchpointBackendResolution.backend.zuoraService.products.filter(_.ratePlanId == formData.ratePlanId).head
```

Comment:
Nearly missed this as the line 63 is so long that it overflows on Github's diff view.

1) are we getting a product or a ratePlan here?... Looks like are getting a Product, but we name the value `ratePlan`
2) The direct call to `.head` is not great as it will produce a very uninformative exception (e.g head called on null object...). If a product without the expected rate plan is definitely a non recoverable circumstance, it still would be better to fail with a slightly more informative message:

```scala
products.filter(_.ratePlanId == formData.ratePlanId).getOrElse(sys.error("Cannot find $"))
```